Cleanup custom ci to build compose app targets (#730)
foundriesio/docs: source/user-guide/custom-ci/custom-ci-for-apps.rst

Custom CI To Build Compose App Targets
======================================

FoundriesFactory® includes all you need to build a containerized application and securely deploy it on devices.
This includes: 

* a git repository
* a CI service that handles the steps to build and delivery apps leveraging the :term:`TUF` compliant OTA service.

Learn more through this :ref:`tutorial <tutorial-compose-app>` on compose apps.

FoundriesFactory consists of well integrated but loosely coupled services.
This allows for using the OTA framework without also using the FoundriesFactory git repos or CI service.
This means you can host your source code anywhere or build your App through any framework, while still leveraging the rest of FoundriesFactory.

This section guides you through the steps of creating a custom CI pipeline in GitHub that:

- builds multi-arch container images and pushes them to the `FoundriesFactory Registry`_;
- builds a `FoundriesFactory Compose App`_ and pushes it to the `FoundriesFactory Registry`_;
- composes `TUF Targets role metadata`_ compliant with our TUF requirements;
- adds the composed TUF Targets to `FoundriesFactory Targets`_.

Prerequisites
-------------

> *  A successful build with a corresponding Target and tag, and the hardware ID of a device for following along

Reviewer: I was a bit confused with this step when following the guide, maybe we should say something like:
* A successful platform build for the corresponding tag (custom-ci-devel)

> *  A successful build with a corresponding Target and tag, and the hardware ID of a device for following along

Author: ```suggestion
*  A successful platform build for the corresponding tag ( ``custom-ci-devel`` in the example), and the hardware ID of a device for following along
```

Prerequisites
-------------

> *  A successful platform build for the corresponding tag ( ``custom-ci-devel`` in the example), and the hardware ID of a device for following along

Reviewer: ```suggestion
*  A successful platform build for the corresponding tag (``custom-ci-devel`` in the example), and the hardware ID (machine) of a device for following along
```

           Tags:          custom-ci-devel
           OSTree Hash:   fe15cf8ad5e09136725ef996c93299d70fa0d20bfa2f10651437b8860b9edcdb

2. `The GitHub repo`_ that contains a working App implementation.

Creating And Setting the Access Token
-------------------------------------

First, you will need to authenticate the GitHub action for the OTA service and the `FoundriesFactory Registry`_.
Create the token to add to the GitHub action via the `FoundriesFactory WebApp`_.

> The token must have ``containers:read-update``, ``targets:read-update`` and ``ci:read-update`` scopes to access the registry and the OTA service.

Reviewer: ```suggestion
The token must have ``containers:read-update``, ``targets:read-update`` and ``ci:read-update`` scopes to access the registry, FoundriesFactory CI and the OTA service.
```
